src/CDSAPI.jl

```diff
@@ -3,6 +3,81 @@ module CDSAPI
 using HTTP
 using JSON
 
+using ScopedValues
+
+
+const key = ScopedValue("")
+const url = ScopedValue("")
```

Comment:
Let's rename these constant variables to `KEY` and `URL` so that they are seen as constants in the codebase and so that they feel more consistent with the env variables, which are also upper case.

This change will also eliminate the `_key` and `_url` naming choice in internal implementations. You can just use `key` and `url` without conflict.